internal/services/watch_btc_events.go

```diff
@@ -174,6 +176,11 @@ func (s *Service) handleSpendingStakingTransaction(
 	// First try to validate as unbonding tx
 	isUnbonding, err := s.IsValidUnbondingTx(spendingTx, delegation, params)
```

Comment:
I think we can simplify `IsValidUnbondingTx` a lot by just comparing the spending tx hash against the hash of `delegation.UnbondingHex`